src/validators/SessionKeyValidator.sol

```diff
@@ -73,11 +67,19 @@ contract SessionKeyValidator is IValidationHook, IModuleValidator, IModule {
   }
 
   function init(bytes calldata data) external {
-    // to prevent recursion, since addHook also calls init
-    if (!_isInitialized(msg.sender)) {
-      IHookManager(msg.sender).addHook(abi.encodePacked(address(this)), true);
-      IValidatorManager(msg.sender).addModuleValidator(address(this), data);
+    // to prevent duplicate inits, since this can be hook plus a validator
+    if (!_isInitialized(msg.sender) && data.length != 0) {
+      require(_addValidationKey(data), "init failed");
+    }
+  }
+
+  function _addValidationKey(bytes memory sessionData) internal returns (bool) {
+    if (sessionData.length == 0) {
+      return false;
     }
```

Comment:
We can now remove this `if` since the validator manager doesn't call `addValidationKey` anymore when adding the validator

Reply:
I thought so too, but the tests totally fail because some of them don't pass a session?!
I was pretty surprised by this, as when I tried to validate adding a session when calling init I also get failures in main (when I require addValidationKey to pass in the add validation module)

Reply:
oh never mind, I avoid this by checking for 0 length in the init function now

Reply:
oh never mind, I avoid this by checking for 0 length in the init function now